subsys/config/src/config_fcb.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <fcb.h>
+#include <string.h>
+
+#include "config/config.h"
+#include "config/config_fcb.h"
+#include "config_priv.h"
+
+#define CONF_FCB_VERS		1
+
+struct conf_fcb_load_cb_arg {
+	load_cb cb;
+	void *cb_arg;
+};
+
+static int conf_fcb_load(struct conf_store *, load_cb cb, void *cb_arg);
+static int conf_fcb_save(struct conf_store *, const char *name,
+			 const char *value);
+
+static struct conf_store_itf conf_fcb_itf = {
+	.csi_load = conf_fcb_load,
+	.csi_save = conf_fcb_save,
+};
+
+int conf_fcb_src(struct conf_fcb *cf)
+{
+	int rc;
+
+	cf->cf_fcb.f_version = CONF_FCB_VERS;
+	cf->cf_fcb.f_scratch_cnt = 1;
+
+	while (1) {
+		rc = fcb_init(CONFIG_CFGSYS_FCB_FLASH_AREA, &cf->cf_fcb);
+		if (rc) {
+			return -EINVAL;
+		}
+
+		/*
+		 * Check if system was reset in middle of emptying a sector.
+		 * This situation is recognized by checking if the scratch block
+		 * is missing.
+		 */
+		if (fcb_free_sector_cnt(&cf->cf_fcb) < 1) {
+
+			rc = flash_area_erase(cf->cf_fcb.fap,
+					cf->cf_fcb.f_active.fe_sector->fs_off,
+					cf->cf_fcb.f_active.fe_sector->fs_size);
+
+			if (rc) {
+				return -EIO;
+			}
+		} else {
+			break;
+		}
+	}
+
+	cf->cf_store.cs_itf = &conf_fcb_itf;
+	conf_src_register(&cf->cf_store);
+
+	return 0;
+}
+
+int conf_fcb_dst(struct conf_fcb *cf)
+{
+	cf->cf_store.cs_itf = &conf_fcb_itf;
+	conf_dst_register(&cf->cf_store);
+
+	return 0;
+}
+
+static int conf_fcb_load_cb(struct fcb_entry_ctx *entry_ctx, void *arg)
+{
+	struct conf_fcb_load_cb_arg *argp;
+	char buf[CONF_MAX_NAME_LEN + CONF_MAX_VAL_LEN + 32];
```

Comment:
32? can you add a macro?

Reply:
replaced by CONF_EXTRA_LEN  - thanks for hint